src/coreclr/vm/ClrEtwAll.man

```diff
@@ -1775,6 +1781,34 @@
                         </UserData>
                     </template>
 
+                    <template tid="WaitHandleWaitStart">
+                        <data name="WaitSource" inType="win:UInt8" map="WaitHandleWaitSourceMap" />
+                        <data name="ClrInstanceID" inType="win:UInt16" />
+                        <data name="AssociatedObjectID" inType="win:Pointer" />
+                        <UserData>
+                            <WaitHandle xmlns="myNs">
+                                <WaitSource> %1 </WaitSource>
+                                <ClrInstanceID> %2 </ClrInstanceID>
+                                <AssociatedObjectID> %3 </AssociatedObjectID>
+                            </WaitHandle>
+                        </UserData>
+                    </template>
+
+                    <template tid="WaitHandleWaitStop">
+                        <data name="WaitSource" inType="win:UInt8" map="WaitHandleWaitSourceMap" />
+                        <data name="ClrInstanceID" inType="win:UInt16" />
+                        <data name="AssociatedObjectID" inType="win:Pointer" />
```

Comment:
The associated object ID is probably not necessary for the Stop event, there would be a matching Stop event for each Start event even in cases involving reentry, which would probably be less common anyway.

Reply:
Ok, I was wondering how a user could match a start from a stop. Maybe that's not a use-case

Reply:
All events are tagged with thread id, so there will be a sequence of start/stop pairs per thread, so a stop should be related to the previous start event on the same thread. It means that the code parsing the nettrace file needs to keep track of start events if there is an interest in the object id and wait source. So exclude object id on stop will make parsing a little more complicated in favor of saving 4-8 bytes per stop event.

Reply:
I figured saving on the size would be preferrable, esp. as these events can be noisy. I believe the parsing in PerfView already keeps track of start/stop events to compute a duration.